homeassistant/components/switch/raspyrfm.py

```diff
@@ -0,0 +1,151 @@
+from homeassistant.const import (
+    CONF_HOST, CONF_NAME, CONF_PORT, CONF_SWITCHES,
+    DEVICE_DEFAULT_NAME)
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['raspyrfm-client==1.2.8']
+_LOGGER = logging.getLogger(__name__)
+
+CONF_GATEWAY_MANUFACTURER = 'gateway_manufacturer'
+CONF_GATEWAY_MODEL = 'gateway_model'
+CONF_CONTROLUNIT_MANUFACTURER = 'controlunit_manufacturer'
+CONF_CONTROLUNIT_MODEL = 'controlunit_model'
+CONF_CHANNEL_CONFIG = 'channel_config'
+DEFAULT_HOST = '127.0.0.1'
+
+# define configuration parameters
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_HOST, default=DEFAULT_HOST): cv.string,
+    vol.Optional(CONF_PORT): cv.port,
+    vol.Optional(CONF_GATEWAY_MANUFACTURER): cv.string,
+    vol.Optional(CONF_GATEWAY_MODEL): cv.string,
+    vol.Required(CONF_SWITCHES, default=[]): vol.Schema([{
+        vol.Optional(CONF_NAME, default=DEVICE_DEFAULT_NAME): cv.string,
+        vol.Required(CONF_CONTROLUNIT_MANUFACTURER): cv.string,
+        vol.Required(CONF_CONTROLUNIT_MODEL): cv.string,
+        vol.Required(CONF_CHANNEL_CONFIG): cv.Dict,
+    }])
+}, extra=vol.ALLOW_EXTRA)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the RaspyRFM switch."""
+    from raspyrfm_client import RaspyRFMClient
+    from raspyrfm_client.device_implementations.controlunit. \
+        controlunit_constants import ControlUnitModel
+    from raspyrfm_client.device_implementations.gateway.manufacturer. \
+        gateway_constants import GatewayModel
+    from raspyrfm_client.device_implementations.manufacturer_constants \
+        import Manufacturer
+
+    # read configuration
+    gateway_manufacturer = config.get(CONF_GATEWAY_MANUFACTURER,
+                                      Manufacturer.SEEGEL_SYSTEME.value)
+    gateway_model = config.get(CONF_GATEWAY_MODEL, GatewayModel.RASPYRFM.value)
+    host = config[CONF_HOST]
+    port = config.get(CONF_PORT)
+
+    switches = config[CONF_SWITCHES]
+
+    # create raspyrfm client
+    raspyrfm_client = RaspyRFMClient()
+
+    # try to get controlunit from client
+    gateway = raspyrfm_client.get_gateway(Manufacturer(gateway_manufacturer),
+                                          GatewayModel(gateway_model), host,
+                                          port)
+
+    switch_entities = []
+    for switch in switches:
+        name = config[CONF_NAME]
+        controlunit_manufacturer = switch[CONF_CONTROLUNIT_MANUFACTURER]
+        controlunit_model = switch[CONF_CONTROLUNIT_MODEL]
+        channel_config = switch[CONF_CHANNEL_CONFIG]
+
+        controlunit = raspyrfm_client.get_controlunit(
+            Manufacturer(controlunit_manufacturer),
+            ControlUnitModel(controlunit_model))
+
+        # convert any key thats not a string into a string (needed for api)
+        channel = {}
+        for key, val in channel_config.items():
+            channel[str(key)] = channel_config[key]
```

Comment:
We already have access to the value of the dict in the iteration.

But we should instead just specify that the keys should be strings in the config schema. So then we don't need this.

Reply:
Ok